tests/integration/test.compaction.js

```diff
@@ -507,6 +507,18 @@ adapters.forEach(function (adapter) {
       });
     });
 
+    it('#8525 - Only compact document with seq > last_seq', async function () {
+      const db = new PouchDB(dbs.name, {auto_compaction: false});
+      const rev = (await db.put({_id: 'foo'})).rev;
+      await db.put({_id: 'foo', _rev: rev});
+      // Hack compaction last_seq to make like a previous compaction already
```

Comment:
Is it possible to actually run the compaction instead of hacking the local doc?  This test seems implementation-specific and might be brittle.

Reply:
Hey @alxndrsn!

Thanks for the quick feedback.

I have thought of three ways to automatically test that compaction is not done from the start anymore:

- Thanks to a performance test that would test the performance gain of the PR. That sounds cumbersome and brittle.
- Thanks to a [JS Proxy](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Proxy) that would be plugged on the document object and check that the compaction method of that object is not called. It doesn't sound straightforward and it's also implementation-specific
- The solution I added to the commit: create a fake `_local/compaction` which is implementation-specific too, indeed, but less than the second solution I think.

The third solution was the best looking for me, and somehow easy to fix if it break one day.
I also doubt a bit that the implementation of the `_local/compaction` will ever change, because in hasn't changed since 2014, and also because it would re-trigger the compaction from the start in all the databases out there, which, from my point of view, would be bad.

Do you have any idea of a test that would not be implementation-specific (because I exhausted the ideas I had in mind)?